### DISCUSSION TITLE
Query String resets unnecessarily on $emit

### DISCUSSION THREAD

**Discussion by ShaikhDanish04** [livewire/livewire#4232] [7. Bugs (archived)]
As livewire brings more power to laravel is it great but here I am facing an issue don't know exactly it is intentional or a bug.

**My Component Architecture is like:**
- Panel
   - Page1
      - ListComponent
      - ManageComponent
         - ChildComponent
         - ChildComponent 
   - Page2
      - ViewComponent
   - Page3

Here my Pages Changes via browser event to Panel Component and it works perfect but,

I have assigned the query string to ListComponent as `page1?list_date=2021-11-17&type=day` and I update the child of Manage Component via $emit it works perfectly not matter from where the event comes.

The issue arises when we I am doing `wire:click="$emitTo('panel','refresh')"` and then in panel I have the listener as `refresh=>$refresh`

So, if I have changed my query string to `page1?list_date=2021-11&type=month` from `page1?list_date=2021-11-17&type=day` it resets back to `page1?list_date=2021-11-17&type=day` which should not happen.

Please Help !!!

**Comment by joshhanley** [comment]
@ShaikhDanish04 what patch version of Livewire are you using?

Also can you share any relevant code from your components?

Thanks!

**Reply by ShaikhDanish04** [reply]
Livewire Patch Version is ^2.7

I have made a basic SPA test code at [livewire-spa-raw](https://github.com/ShaikhDanish04/Livewire-SPA), it holds the necessary code here I recreated the issue it can be located at: -` App\Http\Traits\WithManageUser : Line No. [13,21]`

`

namespace App\Http\Traits;

use App\Models\User;

trait WithManageUser
{
    public $selected_user;
    public $selected_user_id;

    protected $queryStringWithManageUser = [
        'selected_user_id' => ['except' => 'id'], // <----- Query String resets unnecessarily on $emit #4232
    ];


    function setSelectedUser($user_id = null)
    {
        $this->selected_user_id = $user_id;
        $this->selected_user = User::find($user_id);
        $this->emit('userSelected'); // <------- Query String resets unnecessarily on $emit #4232
    }
}

`

**Review UseCase**
- Setup the repo
- Go to the /users create few users and refresh the page
- Select edit for any users from listing (table)
- Then it calls the setSelectedUser FN() in which it sets query_string variable and then the $this->emit('userSelected') event is fired which resets back the query_string

**Comment by ShaikhDanish04** [comment]
Seams like the issue is with the `SupportEvents.php` in Livewire as we can see in `SupportBrowserHistory.php` it maintains the Query String properly but in SupportEvents on `component.dehydrate` the `response->effects` is not maintaining the response path.

**Comment by f-ricci** [comment]
I'm experiencing a similar issue with the `$queryString` parameter.

On the same page I have two components (A, B) that both subscribe to the query string parameter ($search). 

When `Component A` updates the query string `$search` (and therefore the query string in the url) the `Component B` gets out of sync and hold the value which was given on page rendering.

### Example

1. Visits `domain.test/path`
2. `Component A $search = null`
3. `Component B $search = null`
4. `Component A` changes the `$search` value to `"find solutions"`
5. URL changes accordingly to `domain.test/path?search=find solutions`
6. `Component B` doesn't updates the value of `$search` and holds the initial value of `null`
